data/src/main/java/com/microsoft/azure/kusto/data/ClientRequestProperties.java

```diff
@@ -70,25 +77,51 @@ public Long getTimeoutInMilliSec() {
         if (timeoutObj instanceof Long) {
             timeout = (Long) timeoutObj;
         } else if (timeoutObj instanceof String) {
-            timeout = LocalTime.parse((String) timeoutObj).toNanoOfDay() / NANOS_TO_MILLIS;
+            timeout = getMillisFromTimespanString((String) timeoutObj);
         } else if (timeoutObj instanceof Integer) {
             timeout = Long.valueOf((Integer) timeoutObj);
         }
 
         return timeout;
     }
 
+    private long getMillisFromTimespanString(String str) throws TimespanParseException {
+        Matcher matcher = PATTERN.matcher(str);
+        if (!matcher.matches()) {
+            throw new TimespanParseException(str);
+        }
+
+        long millis = 0;
+        String days = matcher.group(1);
+        if (days != null) {
+            millis += Long.parseLong(days) * SECONDS_PER_DAY * 1000L;
+        }
+
+        millis += LocalTime.parse(matcher.group(2)).toNanoOfDay() / 1000000L;
+        return millis;
+    }
+
     public void setTimeoutInMilliSec(Long timeoutInMs) {
         options.put(OPTION_SERVER_TIMEOUT, timeoutInMs);
     }
 
     JSONObject toJson() {
         try {
             JSONObject optionsAsJSON = new JSONObject(this.options);
-            Long timeoutInMilliSec = getTimeoutInMilliSec();
-            if (timeoutInMilliSec != null) {
-                LocalTime localTime = LocalTime.ofNanoOfDay(timeoutInMilliSec * NANOS_TO_MILLIS);
-                optionsAsJSON.put(OPTION_SERVER_TIMEOUT, localTime.toString());
+            Object timeoutObj = getOption(OPTION_SERVER_TIMEOUT);
```

Comment:
Technically not yours, but isn't it weird that we support all of these types in the OPTION_SERVER_TIMEOUT, but we only officially provide one way to set it - setTimeoutInMilliSec?

It's not documented anywhere as far as I know, so we should either add other methods or not support it

Reply:
The documentation I see only mentions allowing a C# Timespan:
https://docs.microsoft.com/en-us/azure/data-explorer/kusto/api/netfx/request-properties
https://docs.microsoft.com/en-us/azure/data-explorer/kusto/concepts/querylimits

Reply:
But why are we providing them so many ways to set the value? Shouldn't we require they stick to the documented approach?